dev/com.ibm.ws.config/src/com/ibm/ws/config/admin/internal/ConfigurationStore.java

```diff
@@ -189,83 +124,61 @@ public synchronized void unbindConfigurations(Bundle bundle) {
         }
     }
 
-    void serializeConfigurationData(File configFile, ExtendedConfigurationImpl config) throws IOException {
-        FileOutputStream fos = null;
-        ObjectOutputStream oos = null;
-        try {
-            // get data outside of the sync block to prevent a deadlock, however we may have inconsistent information
-            Dictionary<String, ?> properties = config.getReadOnlyProperties();
-            Set<ConfigID> references = config.getReferences();
-            Set<String> uniqueVars = config.getUniqueVariables();
-
-            String bundleLocation = config.getBundleLocation();
+    synchronized void saveConfigurationDatas(boolean cancelSaveTask) throws IOException {
+        if (dirty) {
+            List<ExtendedConfigurationImpl> persistConfigs = new ArrayList<>();
+            for (ExtendedConfigurationImpl persistConfig : configurations.values()) {
+                if (persistConfig.getReadOnlyProperties() != null) {
+                    persistConfigs.add(persistConfig);
+                }
+            }
+            ConfigurationStorageHelper.store(persistentConfig, persistConfigs);
+            dirty = false;
+            if (cancelSaveTask && saveTask != null) {
+                saveTask.cancel(false);
+            }
+            saveTask = null;
+        }
+    }
 
-            // write config data
-            fos = new FileOutputStream(configFile, false);
-            oos = new ObjectOutputStream(new BufferedOutputStream(fos));
-            oos.writeObject(properties);
-            oos.writeObject(bundleLocation);
-            oos.writeBoolean(Boolean.FALSE);
-            oos.writeObject(references);
-            oos.writeObject(uniqueVars);
-        } finally {
-            ConfigUtil.closeIO(oos);
-            ConfigUtil.closeIO(fos);
+    private synchronized void loadConfigurationDatas(File configDatas) {
+        if (configDatas.isFile()) {
+            ConfigStorageConsumer<ExtendedConfigurationImpl> consumer = new ConfigStorageConsumer<ExtendedConfigurationImpl>() {
+                @Override
+                public ExtendedConfigurationImpl consumeConfigData(String location, Set<String> uniqueVars, Set<ConfigID> references, ConfigurationDictionary dict) {
+                    String pid = (String) dict.get(Constants.SERVICE_PID);
+                    String factoryPid = (String) dict.get(ConfigurationAdmin.SERVICE_FACTORYPID);
+                    return new ExtendedConfigurationImpl(caFactory, location, factoryPid, pid, dict, references, uniqueVars);
+                }
+            };
+            Collection<ExtendedConfigurationImpl> configs = Collections.emptyList();
+            try {
+                configs = ConfigurationStorageHelper.load(configDatas, consumer);
```

Comment:
Based just on what is happening here it seems that perhaps ConfigurationStorageHelper.load would be more useful if it returned a Map so line 160-162 isn't needed.

Reply:
Done